src/main/java/org/variantsync/diffdetective/diff/difftree/DiffNode.java

```diff
@@ -1112,6 +764,204 @@ public String toTextDiff() {
+            var node = getBackingNode();
+            var current = node.getToLine();
+            node.setToLine(new DiffLineNumber(
+                current.inDiff(),
+                time == BEFORE ? to : current.beforeEdit(),
+                time == AFTER ? to : current.afterEdit()
+            ));
+        }
+
+        @Override
+        public VariationNode<DiffNode> getParent() {
+            var parent = getBackingNode().getParent(time);
+
+            if (parent == null) {
+                return null;
+            } else {
+                return parent.projection(time);
+            }
+        }
+
+
+        @Override
+        public List<VariationNode<DiffNode>> getChildren() {
+            return FilterMappedList.filterMap(
+                getBackingNode().childOrder,
+                (child) -> {
+                    if (getBackingNode().isChild(child, time)) {
+                        return Optional.of(child.projection(time));
+                    } else {
+                        return Optional.empty();
+                    }
+                }
+            );
+        }
+
+        @Override
+        public void addChild(final VariationNode<DiffNode> child) {
+            getBackingNode().addChild(child.getBackingNode(), time);
+        }
+
+        @Override
+        public void addChild(final VariationNode<DiffNode> child, int index) {
+            // The method `DiffNode.addChild` can't be used here because `index` has a different
+            // meaning: For `DiffNode.addChild` it counts all children, before and after, but here
+            // it only counts children at `time`.
+
+            var iterator = getBackingNode().childOrder.listIterator();
+            for (int i = 0; i < index; ) {
+                if (!iterator.hasNext()) {
+                    throw new IllegalArgumentException();
+                }
+
+                if (iterator.next().getDiffType().existsAtTime(time)) {
+                    ++i;
+                }
+            }
+
+            iterator.add(child.getBackingNode());
+        }
+
+        @Override
+        public boolean removeChild(final VariationNode<DiffNode> child) {
+            return getBackingNode().removeChild(child.getBackingNode(), time);
+        }
+
+        @Override
+        public void removeAllChildren() {
+            getBackingNode().removeChildren(time);
+        }
+
+        @Override
+        public Node getDirectFeatureMapping() {
+            return getBackingNode().getDirectFeatureMapping();
+        }
+
+        @Override
+        public int getID() {
+            return getBackingNode().getID();
+        }
+
+        @Override
+        public boolean isSameAs(VariationNode<DiffNode> other) {
+            if (other != null && getClass() == other.getClass()) {
+                Projection otherProjection = (Projection) other;
+                return time.equals(otherProjection.time) && getBackingNode() == otherProjection.getBackingNode();
```

Comment:
Well it's documented that you should use `isSameAs` instead of `==` for identity tests.  In this case it's an implementation detail that `DiffNode.Projection` guarantees that this works and therefore it's documented in `DiffNode.Projection`.

Reply:
ok